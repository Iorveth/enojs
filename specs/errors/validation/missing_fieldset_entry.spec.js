const eno = require('../../../eno.js');

const input = 'emptyness:';

describe('validation.missingFieldsetEntry', () => {
  describe('via Fieldset element', () => {
    const document = eno.parse(input);

    let error;
    try {
      document.fieldset('emptyness').element('presence', { enforceElement: true });
    } catch(err) {
      error = err;
    }

    it('provides a correct message', () => {
      expect(error.message).toMatchSnapshot();
    });

    it('provides correct selection metadata', () => {
      expect(error.selection).toMatchSnapshot();
    });
  });

  describe('via Fieldset entry', () => {
    const document = eno.parse(input);

    let error;
    try {
      document.fieldset('emptyness').entry('presence', { enforceElement: true });
    } catch(err) {
      error = err;
    }

    it('provides a correct message', () => {
      expect(error.message).toMatchSnapshot();
    });

    it('provides correct selection metadata', () => {
      expect(error.selection).toMatchSnapshot();
    });
  });
});
