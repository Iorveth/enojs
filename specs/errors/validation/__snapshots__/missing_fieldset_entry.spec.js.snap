// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation.missingFieldsetEntry via Fieldset element provides a correct message 1`] = `
"The fieldset entry 'presence' is missing - in case it has been specified look for typos and also check for correct capitalization.

   Line | Content
 >    1 | emptyness:
"
`;

exports[`validation.missingFieldsetEntry via Fieldset element provides correct selection metadata 1`] = `
Array [
  Array [
    0,
    10,
  ],
  Array [
    0,
    10,
  ],
]
`;

exports[`validation.missingFieldsetEntry via Fieldset entry provides a correct message 1`] = `
"The fieldset entry 'presence' is missing - in case it has been specified look for typos and also check for correct capitalization.

   Line | Content
 >    1 | emptyness:
"
`;

exports[`validation.missingFieldsetEntry via Fieldset entry provides correct selection metadata 1`] = `
Array [
  Array [
    0,
    10,
  ],
  Array [
    0,
    10,
  ],
]
`;
